src/app/products/charges/create-charge/create-charge.component.ts

```diff
@@ -0,0 +1,215 @@
+          this.chargeForm.addControl('incomeAccountId', new FormControl(''));
+          break;
+        case 4: // Shares
+          this.chargeForm.removeControl('chargePaymentMode');
+          this.chargeForm.removeControl('incomeAccountId');
+          this.chargeForm.get('penalty').setValue(false);
+          break;
+      }
+      this.chargeForm.get('chargeCalculationType').reset();
+      this.chargeForm.get('chargeTimeType').reset();
+    });
+    this.chargeForm.get('chargeTimeType').valueChanges.subscribe((chargeTimeType) => {
+      this.chargeForm.removeControl('feeFrequency');
+      this.chargeForm.removeControl('feeInterval');
+      this.chargeForm.removeControl('feeOnMonthDay');
+      this.chargeForm.removeControl('addFeeFrequency');
+      switch (chargeTimeType) {
+        case 6: // Annual Fee
+          this.chargeForm.addControl('feeOnMonthDay', new FormControl('', Validators.required));
+          break;
+        case 7: // Monthly Fee
+          this.chargeForm.addControl('feeOnMonthDay', new FormControl(''));
+          this.chargeForm.addControl('feeInterval', new FormControl('', [Validators.required, Validators.min(1), Validators.max(12), Validators.pattern('^\\d*$')]));
+          this.repeatEveryLabel = 'Months';
+          break;
+        case 9: // Overdue Fee
+          this.chargeForm.get('penalty').setValue(true);
+          this.chargeForm.addControl('addFeeFrequency', new FormControl(false));
+          this.chargeForm.get('addFeeFrequency').valueChanges.subscribe((addFeeFrequency) => {
+            if (addFeeFrequency) {
+              this.chargeForm.addControl('feeFrequency', new FormControl('', Validators.required));
+              this.chargeForm.addControl('feeInterval', new FormControl('', [Validators.required, Validators.pattern('^\\d*$')]));
+            } else {
+              this.chargeForm.removeControl('feeFrequency');
+              this.chargeForm.removeControl('feeInterval');
+            }
+          });
+          break;
+        case 11: // Weekly Fee
+          this.chargeForm.addControl('feeInterval', new FormControl('', [Validators.required, Validators.pattern('^\\d*$')]));
+          this.chargeForm.get('penalty').setValue(true);
+          this.repeatEveryLabel = 'Weeks';
+          break;
+      }
+    });
+  }
+
+  /**
+   * Submits the charge form and creates charge,
+   * if successful redirects to charges.
+   */
+  submit() {
+    const prevFeeOnMonthDay: Date = this.chargeForm.value.feeOnMonthDay;
+    const monthDayFormat = 'dd MMM';
+    this.chargeForm.patchValue({
+      feeOnMonthDay: this.datePipe.transform(prevFeeOnMonthDay, monthDayFormat)
+    });
+    const charge = this.chargeForm.value;
+    // TODO: Update once language and date settings are setup
+    charge.locale = 'en';
+    charge.monthDayFormat = monthDayFormat;
+    delete charge.addFeeFrequency;
+    if (!charge.taxGroupId) {
+      delete charge.taxGroupId;
```

Comment:
If not deleted, a 'mandatory field' error will be present.

Reply:
ok